Automattic/jetpack
# CRM: Make sure segment counts are updated when tag changes occur

Fixes: https://github.com/Automattic/zero-bs-crm/issues/2690

## Proposed changes:

* This PR adds two `compileSegment` calls within the `addUpdateTagObjLinks` function in `ZeroBSCRM.DAL3.php` - when adding and removing tags. These calls work on all segments (if segments include relevant tags), making sure counts are updated.
* The reason for this is that the counts on the 'Manage Segments' page - `wp-admin/admin.php?page=manage-segments` - was showing out of date counts straight after tag changes had occurred. 

### Other information:


## Jetpack product discussion

https://github.com/Automattic/zero-bs-crm/issues/2690

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Using the [Jetpack Beta plugin](https://github.com/Automattic/jetpack-beta) on a test site with this branch applied or testing locally with the branch checked out, create a few contacts as well as a tag.
* On the manage segments page - `wp-admin/admin.php?page=manage-segments` - create a new segment based on the contact fields 'Has Tag', set to the tag you've created. 
* Add that tag to a couple of contacts via the contact list view page: `/wp-admin/admin.php?page=manage-customers`.
* Refresh or re-visit the 'Manage Segments' page, and the number in the 'Contact Count' field should be correct. If there is no contact count, open the list view and add 'Contact Count' to the page.
* Remove some tags or add more, and the contact count should update accordingly.
* Without this PR applied, the count won't update straight away - you can deactivate or install and activate the Advanced Segments plugin and then try refreshing the page again, and that should trigger the update instead.
